stan/math/prim/prob/double_exponential_lpdf.hpp

```diff
@@ -53,52 +56,57 @@ return_type_t<T_y, T_loc, T_scale> double_exponential_lpdf(
   }
 
   T_partials_return logp(0.0);
-  operands_and_partials<T_y, T_loc, T_scale> ops_partials(y, mu, sigma);
+  operands_and_partials<T_y_ref, T_mu_ref, T_sigma_ref> ops_partials(
+      y_ref, mu_ref, sigma_ref);
 
-  scalar_seq_view<T_y> y_vec(y);
-  scalar_seq_view<T_loc> mu_vec(mu);
-  scalar_seq_view<T_scale> sigma_vec(sigma);
-  size_t size_sigma = stan::math::size(sigma);
-  size_t N = max_size(y, mu, sigma);
+  const auto& y_col = as_column_vector_or_scalar(y_ref);
+  const auto& mu_col = as_column_vector_or_scalar(mu_ref);
+  const auto& sigma_col = as_column_vector_or_scalar(sigma_ref);
 
-  VectorBuilder<true, T_partials_return, T_scale> inv_sigma(size_sigma);
-  VectorBuilder<include_summand<propto, T_scale>::value, T_partials_return,
-                T_scale>
-      log_sigma(size_sigma);
-  for (size_t i = 0; i < size_sigma; i++) {
-    const T_partials_return sigma_dbl = value_of(sigma_vec[i]);
-    inv_sigma[i] = inv(sigma_dbl);
-    if (include_summand<propto, T_scale>::value) {
-      log_sigma[i] = log(sigma_dbl);
-    }
-  }
+  const auto& y_arr = as_array_or_scalar(y_col);
```

Comment:
Actually can you just try `as_array_or_scalar(y_ref)`? I think `as_array_or_scalar` is more general already: https://mc-stan.org/math/dc/d25/as__array__or__scalar_8hpp_source.html vs https://mc-stan.org/math/d8/d8a/as__column__vector__or__scalar_8hpp_source.html

Reply:
We need to convert everything into column vectors (which `as_array_or_scalar` does not do), as distributions accept arbitrary combinations of row and column inputs and Eigen does not do math on combinations of row and column vectors.